### ISSUE TITLE
[Minor] Oversized exponents when nested

### ISSUE THREAD

**Issue by Alex-Programs** [typst/typst#1823]
### Description

Here is a LaTeX formula:
```
\int_{\sum\limits_{n=0}^{\infty}(\sqrt{n}-\sqrt{n+1})}^{\sum\limits_{n=0}^{\infty}(\frac{(-1)^n}{n+1})}(\lim_{t\to\infty}(1+\frac{1}{e^t})^{e^t})^{\frac{d}{dx}(\frac{x^2}{sin^2x+cos^2x})})^2
```

It is rendered by MathJax (inside Obsidian.md) as:
![image](https://github.com/typst/typst/assets/40502404/ac4d33a5-1a5f-46e2-a0ec-59bc1d107c4b)

Here is its typst equivalent:
```
$ (integral^(sum_(n=0)^infinity ((-1)^n)/(n+1))_(sum_(n = 0)^infinity sqrt(n) - sqrt(n+1)) (lim_(t -> infinity) (1 + 1/e^t)^(e^t))^(d/"dx" ( x^2/(sin^x + cos^2x))))^2 $
```

Rendered as:
![image](https://github.com/typst/typst/assets/40502404/9a0e74af-3d25-4c2e-a0dc-00e6069a105e)

That's from the discord bot running on Linux ARM (Pi 4), but I've checked with the latest compiler from github producing a PDF and it shows the same. The screenshot is at the bottom.

Many aspects of the Typst rendering are better, particularly the brackets. However, there are some aspects that aren't as good.

I previously talked about the limits being wrong, but that was my mistake. I've put that at the bottom of the issue.

Exponents are particularly large when nested:
```
$ 5^x^(sin^2(theta)) $
$ sin^2(theta) $
```
![image](https://github.com/typst/typst/assets/40502404/987b17e3-e746-4b3c-bf48-b516b9d9d7f9)

This experiment came out alright:
```
$ integral^(sin^2(theta)) $
$ sum^(sin^2(theta)) $
```
![image](https://github.com/typst/typst/assets/40502404/fd3f30a4-e5e1-46f7-b0aa-529a7a7925dd)

From the offline compiler for Linux. Ignore the coloured background, my windows are partially transparent:
![image](https://github.com/typst/typst/assets/40502404/03a2dc17-9f8f-4b91-a4fe-40138da176ae)

### Operating system

Linux

### Typst version

- [X] I am using the latest version of Typst

---------

---------

I previously wrote:

The rendering of summations in limits - they're rendered to the side, not above and below. I have reproduced this with a more minimal example:
```
$ sum_5^10 $
$ integral^(sum_5^10) $
```
![image](https://github.com/typst/typst/assets/40502404/5c526537-3e78-472c-a034-bb2ff2f39a4d)

There also aren't the requested brackets in the limits - is there a way to disable the auto bracket removal?

**Comment by Enivex** [comment]
> Here is its typst equivalent:

These aren't equivalent, because you're using \limits in tex and not typst

**Comment by Alex-Programs** [comment]
Thank you.

The bug (appears to? I might be using it wrong) still occur when using `attach`:
```
$ attach(sum, t: 10, b: 5) $

$ integral^(attach(sum, t: 10, b: 5)) $
```

I may just be missing the correct way to insert limits.

![image](https://github.com/typst/typst/assets/40502404/4d4e7244-5021-4c9f-bad2-ef4ab0b370e3)

Thank you for the info on the parentheses.

I've been checking against my `cargo install`-d version:

![image](https://github.com/typst/typst/assets/40502404/708bed7f-c955-4e0b-91ba-60e3c04a8b63)

```
$ typst --version
typst 0.6.0 (66df130c)
```
(Which is the most recent commit)

**Comment by Enivex** [comment]
> The bug (appears to? I might be using it wrong) still occur when using `attach`:

It's not a bug. The t and b parameters in attach are smart, and will attach to either top and bottom, or top right and bottom right depending on the operator and context. It's also entirely equivalent to using the `^` and `_` shorthand.

If you want limits even in scripts you need to write `limits(sum)_a^b` instead of sum, just like you write `\sum\limits_a^b` in tex

**Comment by Alex-Programs** [comment]
OK, thanks for the info about the limits.

**Comment by damaxwell** [comment]
About those exponents, here's XeTeX:
<img width="271" alt="image" src="https://github.com/typst/typst/assets/918465/0a2a496c-9020-45c4-bde4-ef8644a5928c">
The primary (only?) difference I see between this and Typst's rendering is that in the Typst version,`sin` isn't in an optically sized variation, it's just the original font scaled down.  Because the 2 is (correctly) optically sized, this highlights the that fact that `sin` is not.

Optical sizing is currently only partly implemented: only single characters and numbers.  It's feature complete as a side benefit of #1779. It could be patched up independently if needed if #1779 is delayed.

**Comment by Alex-Programs** [comment]
Thank you for the information. I'll close this issue.
